adafruit_wiznet5k/adafruit_wiznet5k_socket.py

```diff
@@ -64,8 +65,6 @@ def htons(x):
 AF_INET = const(3)
 NO_SOCKET_AVAIL = const(255)
 
-# keep track of sockets we allocate
-SOCKETS = []
```

Comment:
Why?  And what's supposed to be the interface for this?
The existing `SOCKET` list implementation didn't really do this and didn't in any way prevent sockets that can't exist from being allocated.

The constructor in `adafruit_wiznet5k_socket.py` uses the low level interface's `get_socket`, this just returns 0 if there is no free socket.  But 0 is a valid socket number and all kinds of ugly things happen when you use that afterward (like it may be stomping on the communication of the real owner of socket 0).

This is the way it was, and I don't see a reason to burden this pull request with having to fix everything in this driver.  To be honest, it was (and still is) an ugly and buggy piece of code, and it's going to take more than just this effort to fix it.  I just fixed some bugs I ran into in the process of implementing the server functionality, this is not a general big fix effort.